src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -422,7 +422,7 @@ private Task InvokeAllExceptionFiltersAsync()
 
                     _diagnosticSource.AfterOnExceptionAsync(_exceptionContext, current.FilterAsync);
 
-                    if (_exceptionContext.Exception == null)
+                    if (_exceptionContext.ExceptionHandled)
```

Comment:
Why is this condition different from the other ones? Think what would happen if you set the `.Exception` to null. Would other exception filters run?